api/org/commcare/api/json/PromptToJson.java

```
@@ -0,0 +1,228 @@
+package org.commcare.api.json;
+
+import org.javarosa.core.model.Constants;
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.SelectChoice;
+import org.javarosa.core.model.data.*;
+import org.javarosa.core.model.data.helper.Selection;
+import org.javarosa.core.model.utils.DateUtils;
+import org.javarosa.form.api.FormEntryCaption;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryModel;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Functions for generating the JSON repesentation of a FormEntryPrompt
+ *
+ * @author wspride
+ */
+public class PromptToJson {
+
+    /**
+     * @param prompt       The FormEntryPrompt under consideration
+     * @param questionJson the JSON object question representation being generated
+     */
+    public static void parseQuestion(FormEntryPrompt prompt, JSONObject questionJson) {
+        parseCaption(prompt, questionJson);
+        questionJson.put("help", jsonNullIfNull(prompt.getHelpText()));
+        questionJson.put("binding", jsonNullIfNull(prompt.getQuestion().getBind().getReference().toString()));
+        questionJson.put("style", jsonNullIfNull(parseStyle(prompt)));
+        questionJson.put("datatype", jsonNullIfNull(parseControlType(prompt)));
+        questionJson.put("required", prompt.isRequired() ? 1 : 0);
+        try {
+            parseQuestionAnswer(questionJson, prompt);
+        } catch (Exception e) {
+            e.printStackTrace();
```

What are the failures here and why is it fine to swallow them